### PR TITLE
[CI] Use the same template in both PR build and official build.

### PR DESCRIPTION
**Pull request by liushilongbuaa**
#### Why I did it
Use the same build template in build process.
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -32,27 +32,18 @@ resources:
...
-
-  - template: .azure-pipelines/build-template.yml
-    parameters:
-      platform: mellanox
-      platform_short: mlnx
-      cache_mode: rcache
-      sync_rpc_image: true
-
-  - template: .azure-pipelines/build-template.yml
+  - template: .azure-pipelines/azure-pipelines-build.yml
```
Comment (lguohan) [inline azure-pipelines.yml:40]: fix this first. https://github.com/Azure/sonic-buildimage/pull/7319/files#r613663606

Comment (xumia) [reply azure-pipelines.yml:40]: @liushilongbuaa @lguohan , the change looks good for both PR build and official build.

Comment (liushilongbuaa) [reply azure-pipelines.yml:40]: we have checked this behavior, there is no impact on PR check.

**Comment by xumia** [comment]
@liushilongbuaa , we should not cache sonic images, it only took 1 second to build vs image.
https://dev.azure.com/mssonic/build/_build/results?buildId=11463&view=logs&jobId=cef3d8a9-152e-5193-620b-567dc18af272&j=cef3d8a9-152e-5193-620b-567dc18af272&t=359769c4-8b5e-5976-a793-85da132e0a6f

File: .azure-pipelines/cleanup.yml [hunk 2]
```diff
@@ -10,8 +10,5 @@ steps:
     if sudo [ -f dockerfs/var/run/docker.pid ] ; then
         pid=`sudo cat dockerfs/var/run/docker.pid` ; sudo kill $pid
     fi
-    sudo rm -rf dockerfs
-    sudo rm -rf fsroot
-    username=$(id -un)
-    sudo chown -R ${username}.${username} .
+    sudo rm -rf $(ls -A1)
```
Comment (lguohan) [inline .azure-pipelines/cleanup.yml:13]: is this neccessary? if we delete everything, then we will having some difficult time to debug anything.

Comment (liushilongbuaa) [reply .azure-pipelines/cleanup.yml:13]: neccessary. Without this PR build maybe only need 5min. Because target is already exist.

Comment (liushilongbuaa) [reply .azure-pipelines/cleanup.yml:13]: we can remove clean at the end.

File: .azure-pipelines/azure-pipelines-image-template.yml [hunk 3]
```diff
@@ -45,7 +45,9 @@ jobs:
           ENABLE_DOCKER_BASE_PULL=y make PLATFORM=$(PLATFORM_AZP) PLATFORM_ARCH=$(PLATFORM_ARCH) configure
         displayName: 'Make configure'
     postSteps:
-      - publish: $(System.DefaultWorkingDirectory)/target
+      - script: cp target -r $(Build.ArtifactStagingDirectory)/
```
Comment (liushilongbuaa) [inline .azure-pipelines/azure-pipelines-image-template.yml:48]: Keep same file tree with former build template.

File: .azure-pipelines/cleanup.yml (hunk 2 above)
Comment (liushilongbuaa) [inline .azure-pipelines/cleanup.yml:13]: PR build maybe only need 5mins. Because target existing.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -32,27 +32,18 @@ resources:
...
     parameters:
-      platform: vs
-      platform_short: vs
-      cache_mode: rcache
+      buildOptions: 'USERNAME=admin SONIC_BUILD_JOBS=$(nproc) ${{ variables.VERSION_CONTROL_OPTIONS }}'
```
Comment (qiluo-msft) [inline azure-pipelines.yml:42]: >admin [](http://example.com/codeflow?start=30&length=5)

This is the default value, check `DEFAULT_USERNAME`. why you need to specify it in this PR?

File: .azure-pipelines/azure-pipelines-image-template.yml (hunk 3 above)
Comment (qiluo-msft) [reply .azure-pipelines/azure-pipelines-image-template.yml:48]: What is "former build template"? I don't understand the motiviation and why it is required.

Comment (liushilongbuaa) [reply .azure-pipelines/azure-pipelines-image-template.yml:48]: Now the published artifacts is short of a directory 'target'. And we don't want to publish source code.

File: azure-pipelines.yml (hunk 4 above)
Comment (liushilongbuaa) [reply azure-pipelines.yml:42]: It is first used in Guohan's PR pipeline. Maybe there is some corner case, so I kept it.

**Review by lguohan — APPROVED** [review]

**Pull request merged**